internal/payments/service.go

```diff
@@ -0,0 +1,218 @@
+type Service interface {
+	CreateCustomer(ctx context.Context, params CreateCustomerParams) (*stripe.Customer, error)
+	GetCustomer(ctx context.Context, customerID string) (*stripe.Customer, error)
+	GetCustomerDefaultPaymentMethod(ctx context.Context, customerID string) (*stripe.PaymentMethod, error)
+	GetCustomerPaymentsMethods(ctx context.Context, customerID string) ([]*stripe.PaymentMethod, error)
+	GetPaymentMethod(ctx context.Context, id string) (*stripe.PaymentMethod, error)
+	CreateProjectInvoice(ctx context.Context, params CreateProjectInvoiceParams) (*stripe.Invoice, error)
+	GetInvoice(ctx context.Context, invoiceID string) (*stripe.Invoice, error)
+}
+
+// StripeService works with stripe network through stripe-go client
+type StripeService struct {
+	log    *zap.Logger
+	client *client.API
+}
+
+// CreateCustomerParams contains info needed to create new stripe customer
+type CreateCustomerParams struct {
+	Email       string
+	Name        string
+	Description string
+	SourceToken string
+}
+
+// CreateProjectInvoiceParams contains info needed to create project invoice
+type CreateProjectInvoiceParams struct {
+	ProjectName     string
+	CustomerID      string
+	PaymentMethodID string
+
+	Storage     float64
+	Egress      float64
+	ObjectCount float64
+
+	StartDate time.Time
+	EndDate   time.Time
+}
+
+// NewService creates new instance of StripeService initialized with API key
+func NewService(log *zap.Logger, apiKey string) *StripeService {
+	stripe.DefaultLeveledLogger = wrapLogger(log)
+
+	sc := new(client.API)
+	sc.Init(apiKey, nil)
+
+	return &StripeService{
+		log:    log,
+		client: sc,
+	}
+}
+
+// CreateCustomer creates new customer from CustomerParams struct
+// sets default payment to one of the predefined testing VISA credit cards
+func (s *StripeService) CreateCustomer(ctx context.Context, params CreateCustomerParams) (*stripe.Customer, error) {
+	cparams := &stripe.CustomerParams{
+		Email:       stripe.String(params.Email),
+		Name:        stripe.String(params.Name),
+		Description: stripe.String(params.Description),
+	}
+
+	// Set default source (payment instrument)
+	//if params.SourceToken != "" {
+	//	err := cparams.SetSource(params.SourceToken)
+	//	if err != nil {
+	//		return nil, StripeErr.Wrap(err)
+	//	}
+	//}
+
+	// TODO: delete after migrating from test environment
+	err := cparams.SetSource("tok_visa")
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Customers.New(cparams)
+}
+
+// GetCustomer retrieves customer object from stripe network
+func (s *StripeService) GetCustomer(ctx context.Context, id string) (*stripe.Customer, error) {
+	return s.client.Customers.Get(id, nil)
+}
+
+// GetCustomerDefaultPaymentMethod retrieves customer default payment method from stripe network
+func (s *StripeService) GetCustomerDefaultPaymentMethod(ctx context.Context, customerID string) (*stripe.PaymentMethod, error) {
+	cus, err := s.client.Customers.Get(customerID, nil)
+	if err != nil {
+		return nil, err
```

Comment:
Should we have a `paymentErr` to wrap this?

Reply:
I am not sure

Reply:
I like that idea, the more clear the log messages, the better!